### DISCUSSION TITLE
выбор языка

### DISCUSSION THREAD

**Discussion by yurikfm** [petrovvlad/freeLib#158] [General]
Здравствуйте.
На ubuntu 24.04 есть ошибка.
Скрин:
![Screenshot_20240710_194340](https://github.com/petrovvlad/freeLib/assets/142584785/134e4cd1-7411-4b9e-af57-6f54ba533432)
Не могу выбрать язык книги. Точнее, не показывает какой язык выбирается.

**Comment by petrovvlad** [comment]
Если включить колонку языков книг, есть ли у книг язык или колонка остаётся пустой?

**Comment by yurikfm** [comment]
колонка так же остается пустой.  не смог заскринить.
там ошибка в отображении просто. ибо языки выбираются.

**Reply by petrovvlad** [reply]
Похоже у Вас в базе не присвоились языки книгам. Как вы добавляли книги в библиотеку? С inpx файлом или без?

**Comment by yurikfm** [comment]
Как обычно, через файл inpx.

**Reply by petrovvlad** [reply]
Можете дать ссылку на inpx файл, для анализа?

**Comment by yurikfm** [comment]
Сюда нельзя же ...
Залил на диск https://drive.google.com/file/d/1wYfPSsD8C97rFH15pajTvo_nqoqAgIhf/view?usp=sharing

**Comment by yurikfm** [comment]
Дал разрешение на скачку файла. Сбой был какой то что ли ...
Разрешение сразу дал при получении ссылки, но не приняло почему то (((

**Reply by petrovvlad** [reply]
Доступ получил. Я сейчас в отпуске, продолжим позже.

**Comment by petrovvlad** [comment]
Добавил себе библиотеку из вашего inpx файла. У меня языки книги прописались нормально. Вы пересоздать библиотеку?
Если ничего не изменилось скиньте мне вашу базу - файл ~/.local/share/freeLib/freeLib.sqlite.

**Comment by yurikfm** [comment]
Я сделал так:
1. обе версии QT установил. 5 и 6 версию.
2. обновил код src командой `git clone --recurse-submodules https://github.com/petrovvlad/freeLib.git`
3. переустановил программу.
И после переустановки, в терминале указало обновление для 4-5 папок. В результате в FreeLib появился выбор языка.
Некоторое время назад было много обновлений для системы.
Теперь непонятно что именно решило мою проблему.

**Reply by petrovvlad** [reply]
Хорошо, что проблема решилась. Плохо, что мы не знаем причину проблемы.

**Comment by yurikfm** [comment]
А какой именно QT нужен?
Ибо первый раз QT-6 ставил.

**Reply by petrovvlad** [reply]
Рекомендую Qt6. Если стоят обе версии то при сборке указать -DFREELIB_QT_MAJOR_VERSION=6.

**Reply by yurikfm** [reply]
При сборке была одна только версия QT. Но теперь их два ...
А если явно не указывать версию при сборке? Возникнут проблемы?
П.С. Если честно, то не понимаю применение новой версии QT-6. Собиралось и на ранних версиях ...

**Reply by petrovvlad** [reply]
Проблем не должно быть. Для работы opds сервера теперь нужна Qt≥6.4.

**Comment by yurikfm** [comment]
UP:
Приветствую.
Данная проблема возникла уже на минт 22(wilma). Но ситуация ещё хуже, скрин:
![Снимок экрана от 2024-11-20 20-08-57](https://github.com/user-attachments/assets/20d0ee6d-9585-4aeb-b960-27de4ab26361)
Если смотреть на "строку" язык книги, то видно что есть пустые окна.

**Reply by yurikfm** [reply]
Устанавливалось с репо минта.

**Reply by petrovvlad** [reply]
Попробуйте установить libqt5svg5.

**Reply by yurikfm** [reply]
Так он установлен. Пишет: 

> Уже установлен пакет libqt5svg5 самой новой версии (5.15.13-1).

Сейчас попробуем установить версию 5.15.15 ...

**Reply by yurikfm** [reply]
Новая версия не хочет устанавливаться, зависимостей полно ...
Но я поставил компоненты:

> sudo apt update && sudo apt install git cmake build-essential qt6-base-dev libqt6core5compat6-dev qt6-httpserver-dev libqt6websockets6-dev libqt6svg6 libquazip1-qt6-dev libtbb-dev qtkeychain-qt6-dev

И теперь хоть что то показывает. Вот прям как в начале темы на скрине, один в один.
Сама система где проблемы:
`System:
  Kernel: 6.8.0-49-generic arch: x86_64 bits: 64 compiler: gcc v: 13.2.0 clocksource: tsc
  Desktop: Cinnamon v: 6.2.9 tk: GTK v: 3.24.41 wm: Muffin v: 6.2.0 vt: 7 dm: LightDM v: 1.30.0
    Distro: Linux Mint 22 Wilma base: Ubuntu 24.04 noble
`

**Comment by yurikfm** [comment]
Удалил этот минт, поставил  ubuntu 24.10 и собрал из исходников. Собирал по инструкции с главной страницы.
Программа работает хорошо, всё показывает и раздает книги.
Это первое впечатление, будущее покажет.
